### ISSUE TITLE
MediaStream 'ended' event deprecation

### ISSUE THREAD

**Issue by tgabi333** [otalk/localmedia#19]
I've just run into this:

`
The MediaStream 'ended' event is deprecated and will be removed in M53, around September 2016. Please use the MediaStream 'inactive' event instead. See https://www.chromestatus.com/features/5730404371791872 for more details.
`

Do you have any idea how to proceed?
Regarding to firefox:
https://github.com/otalk/localmedia/commit/cd5139182c23e412b945f989d8dd37101a44d7f5

how can i help?

**Comment by tgabi333** [comment]
@fippo Do you think dropping `webrtcsupport` library in favor of `adapter.js`? 

I would like to work on a new version because of the above issue. It could be a new major...

**Comment by fippo** [comment]
away  with webrtcsupport...

**Comment by tgabi333** [comment]
and what about `autoAdjustMic` setting and `mediastream-gain` ?

**Comment by samsont** [comment]
In Chrome 62, in Screen Share, the 'ended' event is not firing.

Line 153 of  localmedia.js, 
  The function  track.addEventListener('ended', function () {
})

is never called. Any thoughts on workarounds?

**Comment by samsont** [comment]
If you click on the system popup which typcially reads: "... is sharing a window .. Stop Sharing/Hide", it works. If the user clicks on a button and you call stopScreenShare(), it doesn't work.

The first fix I have done is to just add another function to detect chrome 62 next to the FireFox fix. That seems to work. Though, it is not satisfying.

**Comment by fippo** [comment]
@samsont if you call stopScreenShare and that calls track.stop() then track.onended will not be called. That is a general design principle for those event handlers. It should not have changed in Chrome 62 though.

**Comment by samsont** [comment]
I agree. My memory is that track.onended is not called. But, the event 'ended' is generated.  And, we are still using that event in this code. It works pretty well for disconnecting streams and the like. It even works for screenshare when using the system popup to  stop the screenshare. But, hitting a button and telling it to stop no longer works.   

Do you have any ideas for replacements for looking for the 'ended' event?

In chrome 62, there is a small note about background video track optimization using media extensions. https://developers.google.com/web/updates/2017/09/chrome-62-media-updates. I don't think it is related -- but it might be.

**Comment by stiv-yakovenko** [comment]
<img width="960" alt="ffproblem" src="https://user-images.githubusercontent.com/4163254/34063968-3a391c54-e228-11e7-8b5f-df489853aed1.png">

Hi! Code contains call to addEventListener. This function is unavailable on my FF (updated to lastest version), so this produces crash, see my screenshot.
